app/src/main/java/org/wikipedia/page/linkpreview/LinkPreviewDialog.kt

```diff
@@ -248,9 +251,14 @@ class LinkPreviewDialog : ExtendedBottomSheetDialogFragment(), LinkPreviewErrorV
         }
     }
 
+    @SuppressLint("RequiresFeature")
     private fun setPreviewContents(contents: LinkPreviewContents) {
         if (!contents.extract.isNullOrEmpty()) {
-            binding.linkPreviewExtract.text = StringUtil.fromHtml(contents.extract)
+            binding.linkPreviewExtractWebview.setBackgroundColor(Color.TRANSPARENT)
+            binding.linkPreviewExtractWebview.setLayerType(WebView.LAYER_TYPE_SOFTWARE, null)
+            // Color hex code from android:textColorPrimary
+            val colorHex = if (WikipediaApp.instance.currentTheme.isDark) "#f8f9fa" else "#202122"
```

Comment:
These values should not be hard-coded here, and can be computed from the actual color resource.

Reply:
I tried to use `Integer.toHexString()` but the regular 6 digits hex string will have an additional `ff` at the beginning of the text after converting it. I use `substring()` to start the index from the 3rd text in order to display the color correctly. Please let me know if you have a better solution for that, thanks!